### ISSUE TITLE
Spelling mistakes should be warnings to prevent getting in the way of tasks/debugging

### ISSUE THREAD

**Issue by Tyriar** [Microsoft/vscode-spell-check#34]
I try to build and get this:

![image](https://cloud.githubusercontent.com/assets/2193314/15809551/7b2d746e-2b48-11e6-9055-0c066b5db315.png)

However the only errors are from my README.md file, obviously it's not going to hurt debugging:

![image](https://cloud.githubusercontent.com/assets/2193314/15809555/9c482ab8-2b48-11e6-81da-ebed7c2bf31d.png)

This forces an extra step in debugging or I can restart vscode (and not open README.md) in order to avoid this error message for spelling errors.

These should probably all be warnings, they're primarily to do with technical terms that teacher doesn't understand, URL fragments and package names.

**Comment by seanmcbreen** [comment]
Yes it's a little unfortunate [the error warning] at the same time a spelling mistake is clearly and error :)  But not one I'd block debugging on :)  Errors get red underlines so I like them for the mainly MD editor crew.  

The good news is this is a simple config away...

File: `spell.json`

```
    "mistakeTypeToStatus": {
        "Passive voice": "Error",
        "Spelling": "Error",
        "Complex Expression": "Warning",
        "Hidden Verbs": "Information",
        "Hyphen Required": "Error",
        "Redundant Expression": "Information",
        "Did you mean...": "Information",
        "Repeated Word": "Error",
        "Missing apostrophe": "Error",
        "Cliches": "Warning",
        "Missing Word": "Warning",
        "Make I uppercase": "Error"
    }
``` 

For the link checking issue the same approach works [a `spell.json`] setting.  This one is pretty aggressive:
* all code blocks
* all image names 
* key binding codes (it's the one I used for our docs)
* Any URL which was the ask :)

> Note: JSON regex are super ugly and need \\\\ for a single \ #notMyFault

```
"ignoreRegExp": [
        "/`(kb.*?)`/g",
        "/(http\\\\S*)/gm",
        "/\\\\(.*.png\\\\)/g",
        "/\\\\(.*.gif\\\\)/g",
        "/\\\\(.*.md\\\\)/g",
        "/\\\\(.*.jpg\\\\)/g",
        "/^((`{3}\\\\s*)(\\\\w+)?(\\\\s*([\\\\w\\\\W]+?)\\\\n*)\\\\2)\\\\n*(?:[^\\\\S\\\\w\\\\s]|$)/gm"
    ]
```

Closing unless u disagree - then re-open :)

**Comment by seanmcbreen** [comment]
Do you mean more than the settings file?  You can at your discretion make them as low as `information`.  

This extension is really for doc authors not coders (as the primary audience) so spelling mistakes (and other things like repeated words) are indeed errors.

Is the issue the default behavior (which is errors):
https://github.com/Microsoft/vscode-spell-check/blob/master/extension.ts#L146

Or that you can't change it?

**Comment by Tyriar** [comment]
@seanmcbreen if it's for doc authors then that makes sense, but it makes the extension an annoyance if you're a dev. And I'd wager the majority of the users using this extension also code.
